src/osd/scrub_machine.cc

```diff
@@ -60,6 +60,12 @@ bool ScrubMachine::is_reserving() const
   return state_cast<const ReservingReplicas*>();
 }
 
+bool ScrubMachine::is_accepting_updates() const
+{
+  // note that this specific test is Primary-specific
```

Comment:
Replace comment with assert(is_primary());

Reply:
Not in the interface currently available to the state machine.
I'd rather not add it just for this assert

Reply:
I think it's worth it.  Quite a few things in this portion of the code base rely on being queued only for replicas or primaries.  Here, you felt it was important enough to warrant a comment.  I think being able to tell dynamically if this constraint is violated is worth a bit of interface bloat.

Reply:
Will fix.